internal/js/summary.js

```diff
@@ -385,7 +387,7 @@ class ReportBuilder {
 		return [
 			...this._renderChecks(group.checks, renderContext),
 			...this._renderMetrics(group.metrics, renderContext),
-			...(group.groups ? this._renderNestedGroups(group.groups) : []),
+			...(group.groups ? this._renderNestedGroups(group.groups, group.groups_order, renderContext) : []),
```

Comment:
The missing `renderContext` here is what was making nested groups to not be properly indented.